packages/react/src/app-layout/utils.ts

```diff
@@ -1,4 +1,38 @@
+import { AppLayoutSidebarNavProps } from './AppLayoutSidebarNav';
+
 // The breakpoint where we show the sidebar in a two column layout
 export const APP_LAYOUT_DESKTOP_BREAKPOINT = 'xl';
 
 export const APP_LAYOUT_SIDEBAR_WIDTH = '16rem'; // 256 px
+
+export const findBestMatch = (
+	items: AppLayoutSidebarNavProps['items'],
+	activePath?: string
+) => {
+	if (!activePath) return '';
+
+	const allHrefs = items
+		.flatMap((item) => (Array.isArray(item) ? item : [item, ...item.items]))
```

Comment:
Again, we should make this consistent with side nav. Not sure which is more performant. That one doesn't seem to iterate if the whole array 3 times though

Reply:
I'm all for that and have tried to keep them as close as possible. AppLayout has to deal with different shaped items and "links" - i.e. items can be objects (for groups) or arrays, and "links" can be links or buttons. Ideally we move to a more composable format to fall inline with the rest of the AppLayout API and remove the button affordance but that would be a breaking change.

I'm open to thoughts on this, mind you.

Reply:
This part seems similar to side nav's `flattenItems` though? It's ok if they're different because of implementation, but flattening the arrays in different ways feels weird

Reply:
Here we're only flattening to two levels and we have to map both levels differently to handle the sometimes-an-object, sometimes-an-array first step. I don't see a benefit with recursion here so I think this is ok.